cpp/memilio/math/integrator.h

```diff
@@ -126,13 +177,20 @@ class OdeIntegrator
             }
             dt_copy = dt;
 
+            if (tmax - t < dt_min_copy) {
+                // reduce minimal step size low enough such that we do not step past tmax
+                m_core->get_dt_min() = dt; // == tmax - t
+                // the following step will be the last. dt_min must be reset after this loop.
+            }
+
             results.add_time_point();
             step_okay &= m_core->step(f, results[i], t, dt, results[i + 1]);
             results.get_last_time() = t;
 
             // if dt has been changed (even slighly) by step, register the current m_core as adaptive
             m_is_adaptive |= !floating_point_equal(dt, dt_copy);
         }
+        m_core->get_dt_min() = dt_min_copy; // reset dt_min
         // if dt was decreased to reach tmax in the last time iteration,
         // we restore it as it is now probably smaller than required for tolerances
         dt = max(dt, dt_restore);
```

Comment:
By resetting/restoring, we get back a dt, which does not reflect the true step size in the last time step.

Reply:
I don't think that is a problem. Adaptive steppers are expected to change dt anyways, and the "true" value is in the results.
The problem this line addresses is that, e.g., we have used dt~1e-3, but dt=tmax-t may have set dt~1e-8 or so. If we restart the integrator, this would cause several steps that are a lot smaller than needed. This is important for graph sims which pause integration frequently.